conda/recipes/libcudf/build.sh

```diff
@@ -1,5 +1,8 @@
 #!/bin/bash
-# Copyright (c) 2018-2022, NVIDIA CORPORATION.
+# Copyright (c) 2018-2023, NVIDIA CORPORATION.
 
 export cudf_ROOT="$(realpath ./cpp/build)"
-./build.sh -n -v libcudf libcudf_kafka benchmarks tests --build_metrics --incl_cache_stats --cmake-args=\"-DCMAKE_INSTALL_LIBDIR=lib -DCUDF_ENABLE_ARROW_S3=ON\"
+./build.sh -n -v \
+    libcudf libcudf_kafka benchmarks tests \
+    --build_metrics --incl_cache_stats \
+    --cmake-args=\"-DCMAKE_INSTALL_LIBDIR=lib -DCUDF_ENABLE_ARROW_S3=ON -DNVBench_ENABLE_CUPTI=OFF\"
```

Comment:
We must disable the NVBench option for cupti in our conda packages of `libcudf-tests` because cupti breaks minor version compatibility. If we build with CUDA 11.8 and cupti enabled, the benchmarks can only be run with CUDA 11.8. I decided that being able to run benchmarks with other CUDA runtimes (of the same major version) is important enough to disable the additional metrics provided by cupti. If those metrics are needed, libcudf benchmarks can be built from source or cupti support can be enabled for a specific build of the `libcudf-tests` package by changing this flag.

Reply:
Disabling this option also simplifies the CUDA 12 conda packaging because it means we don't have to depend on cupti. Using cupti was a pain point during early tests of CUDA 12 conda builds: https://github.com/rapidsai/cudf/pull/12922#issuecomment-1478359438